protected/humhub/modules/content/Events.php

```diff
@@ -143,6 +144,27 @@ public static function onCronBeforeAction($event): void
         }
     }
 
+    public static function onViewEndBody($event)
+    {
+        $highlight = Yii::$app->session->get('contentHighlight');
+        if ($highlight === null || $highlight === '') {
+            return;
+        }
+        Yii::$app->session->remove('contentHighlight');
+
+        /* @var View $view */
+        $view = $event->sender;
+
+        $index = time();
+        $view->registerJs('function contentHighlight' . $index . '() {
```

Comment:
Couldn't we somehow outsource this JS to e.g. the `search` or `content` javascript?

Reply:
commit https://github.com/humhub/humhub/pull/6971/commits/55381159b7a3ad050ebd6d02aa56a7bff41c7733